File: Data Structures & Algorithms/duplicate-integer/submission-9.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        // Create an unordered map to store each number
        std::unordered_map<int, bool> nMap; 

        // Check each number
        for(int n : nums) {
            // check if the hash map contains the value
            if(nMap.contains(n)) {
                // already in the map, so false
                return true; 
            }
            // set the value to true
            nMap[n] = true; 
        }
        // no dups found 
        return false; 
    }
};

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-9.cs
public class Solution {
    public int RemoveDuplicates(int[] nums) {
        // Check edge cases
        if(nums.Length == 0) 
        {
            return 0;
        }

        if(nums.Length == 1) {
            return 1; 
        }
        int l = 1; 
        for(int r =1; r < nums.Length; r++ ) {
            if(nums[r] != nums[l - 1]) {
                nums[l] = nums[r];
                l++;
            }
        }

        return l;
    }
}